apps/examples/src/misc/diff.ts

```diff
@@ -0,0 +1,23 @@
+function format(prefix: string, key: string) {
+	if (prefix && prefix !== '') {
+		return `${prefix}/${key}`
+	}
+	return key
+}
+
+export function getDiff(prev: any, next: any, prefix = '', diff: any = {}) {
+	const allKeys = new Set([...Object.keys(prev), ...Object.keys(next)])
+	for (const key of allKeys) {
+		const prevValue = prev[key]
+		const nextValue = next[key]
+		const keyPath = format(prefix, key)
+		if (prevValue !== nextValue) {
+			if (typeof prevValue === 'object' && typeof nextValue === 'object') {
+				getDiff(prevValue, nextValue, keyPath, diff)
+			} else {
+				diff[keyPath] = { prev: prev[key], next: next[key] }
```

Comment:
i think a little tweak here would be to add the shape id to the table (otherwise if there's multiple things being tracked it's hard to tell what's what.

this isn't perfect, but you could do:
```
diff[keyPath] = { [`prev:${prev.id}`]: prev[key], [`next:${next.id}`]: next[key] }
```